ios/brave-ios/Sources/Brave/Frontend/Browser/Toolbars/BottomToolbar/Menu/Bookmarks/BookmarksView.swift

```diff
@@ -0,0 +1,531 @@
+                    model.handleBookmarkItemSelection(.openInNewPrivateTab, node: node)
+                  } label: {
+                    Label(
+                      Strings.openNewPrivateTabButtonTitle,
+                      systemImage: "plus.square.fill.on.square.fill"
+                    )
+                  }
+                }
+              }
+
+              Section {
+                Button {
+                  model.handleBookmarkItemSelection(.copyLink, node: node)
+                } label: {
+                  Label(Strings.copyLinkActionTitle, systemImage: "doc.on.doc")
+                }
+
+                Button {
+                  model.handleBookmarkItemSelection(.shareLink, node: node)
+                } label: {
+                  Label(Strings.shareLinkActionTitle, systemImage: "square.and.arrow.up")
+                }
+              }
+            }
+          }
+      }
+      .onMove { (indexSet, newOffset) in
+        let nodes = viewModel.items
+        let nodesToMove = indexSet.map({ nodes[$0] })
+        model.move(nodes: nodesToMove, to: newOffset)
+      }
+      .onDelete { indexSet in
+        withAnimation {
+          let nodes = viewModel.items
+          let nodesToDelete = indexSet.map({ nodes[$0] })
+          model.delete(nodes: nodesToDelete)
+        }
+      }
+    }
+    .scrollContentBackground(.hidden)
+    .background(Color(.braveGroupedBackground))
+    .listStyle(.plain)
+    .environment(\.editMode, $editMode)
+    .navigationBarTitleDisplayMode(.inline)
+    .navigationTitle(viewModel.folder?.title ?? Strings.bookmarks)
+    .toolbarBackground(.visible, for: .navigationBar)
+    .overlay {
+      if viewModel.items.isEmpty {
+        BookmarksEmptyStateView(isSearching: !searchText.isEmpty)
+      }
+    }
+    .toolbar {
+      ToolbarItemGroup(placement: .confirmationAction) {
+        Button {
+          dismiss = true
+        } label: {
+          Text(Strings.done)
+        }
+      }
+
+      ToolbarItemGroup(placement: .bottomBar) {
+        Button {
+          if viewModel.folder == nil {
+            showShare = true
+          } else {
+            showCreateFolder = true
+          }
+        } label: {
+          Label {
+            Text(Strings.newFolderTitle)
+          } icon: {
+            Image(braveSystemName: viewModel.folder == nil ? "leo.share.macos" : "leo.folder.new")
+              .foregroundStyle(Color(braveSystemName: .iconInteractive))
+          }
+          .labelStyle(.iconOnly)
+        }
+
+        Spacer()
+
+        if viewModel.folder != nil && !viewModel.items.isEmpty {
+          Button {
+            editMode = editMode.isEditing ? .inactive : .active
+          } label: {
+            Label {
+              Text(Strings.editBookmarkTitle)
+            } icon: {
+              Image(braveSystemName: "leo.edit.pencil")
+                .foregroundStyle(Color(braveSystemName: .iconInteractive))
+            }
+            .labelStyle(.iconOnly)
```

Comment:
I realize this is the same as whats shipping but really we should be updating the icon based on the current state, or at the very least the text that's read out in VoiceOver to specify that tapping will _leave_ edit mode

Reply:
Fixed. Added two strings, so now the button's voice-over will change depending on the button state.